docs/usage/shoot_pod_autoscaling_best_practices.md

```diff
@@ -0,0 +1,327 @@
+# Introduction
```

Comment:
Hmm... I tried to follow "the style" in this repo.

I didn't see ToCs in any other document (just grepped; maybe I missed it?), including the other large guide https://github.com/gardener/gardener/blob/master/docs/usage/shoot_high_availability_best_practices.md (arguably, that I wrote, but that everybody reviewed).

I also don't know whether this would ever conflict with DocForge or any other tool "generating" a ToC on the fly.

Reply:
github automatically generates a TOC based on the headlines, so if you really need it, you can switch it on with the click of a button. Shouldn't this be enough? 
<img width="1264" alt="Screenshot 2024-07-16 at 09 06 32" src="https://github.com/user-attachments/assets/954562cd-6ac1-48ba-85a9-6ce78e44b212">

Reply:
Thanks!
The github generated table of content will do the work, let's keep it as it is then.

Reply:
Thanks @voelzmo!